actor/src/main/scala-2.12/org/apache/pekko/util/ByteIterator.scala

```diff
@@ -470,7 +470,7 @@ abstract class ByteIterator extends BufferedIterator[Byte] {
   override def indexWhere(p: Byte => Boolean): Int = indexWhere(p, 0)
   override def indexWhere(p: Byte => Boolean, from: Int): Int = {
     var index = 0
-    while (index < from) {
+    while (index < from && hasNext) {
```

Comment:
if there is a bug in ByteIterator, could the bug fix be in a separate PR? It needs to have its own release note and may be backported to 1.0.x,

Reply:
Extracted in https://github.com/apache/pekko/pull/1282

I'll cleanup the history here in a bit, to avoid potential merge conflicts.